### ISSUE TITLE
Jute Seed Crash - 1.10.2

### ISSUE THREAD

**Issue by MilchLos** [squeek502/VeganOption#81]
Everytime I plant a Jute seed on a grass block my game crashes. Not only grass but dirt and farmland as well. I need to test it once more.

[crash-2017-07-04_20.55.58-client.txt](https://github.com/squeek502/VeganOption/files/1122860/crash-2017-07-04_20.55.58-client.txt)

edit by @elifoster July 4 2017 4:43 PM PST:
```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  ASMPlugin (VeganOption-mc1.10.2-0.2.0.jar)
Contact their authors BEFORE contacting forge

// Ouch. That hurt :(

Time: 7/4/17 8:55 PM
Description: Exception while adding particle

java.lang.NullPointerException: Exception while adding particle
	at net.minecraft.world.biome.BiomeColorHelper.func_180285_a(SourceFile:35)
	at net.minecraft.world.biome.BiomeColorHelper.func_180286_a(SourceFile:47)
	at squeek.veganoption.blocks.BlockJutePlant$ColorHandler.func_186720_a(BlockJutePlant.java:240)
	at net.minecraft.client.renderer.color.BlockColors.func_186724_a(BlockColors.java:167)
	at net.minecraft.client.particle.ParticleDigging.func_187154_b(SourceFile:52)
	at net.minecraft.client.particle.ParticleDigging.func_174845_l(SourceFile:47)
	at net.minecraft.client.particle.ParticleDigging$Factory.func_178902_a(SourceFile:105)
	at net.minecraft.client.particle.ParticleManager.func_178927_a(ParticleManager.java:135)
	at net.minecraft.client.renderer.RenderGlobal.func_174974_b(RenderGlobal.java:2132)
	at net.minecraft.client.renderer.RenderGlobal.func_180442_a(RenderGlobal.java:2086)
	at net.minecraft.world.World.func_175720_a(World.java:1092)
	at net.minecraft.world.World.func_175688_a(World.java:1079)
	at net.minecraft.entity.Entity.func_174808_Z(Entity.java:1146)
	at net.minecraft.entity.Entity.func_174830_Y(Entity.java:1132)
	at net.minecraft.entity.Entity.func_70030_z(Entity.java:436)
	at net.minecraft.entity.EntityLivingBase.func_70030_z(EntityLivingBase.java:216)
	at net.minecraft.entity.Entity.func_70071_h_(Entity.java:359)
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:1932)
	at net.minecraft.entity.player.EntityPlayer.func_70071_h_(EntityPlayer.java:233)
	at net.minecraft.client.entity.EntityPlayerSP.func_70071_h_(EntityPlayerSP.java:167)
	at net.minecraft.world.World.func_72866_a(World.java:1964)
	at net.minecraft.world.World.func_72870_g(World.java:1934)
	at net.minecraft.world.World.func_72939_s(World.java:1750)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1789)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at net.minecraft.world.biome.BiomeColorHelper.func_180285_a(SourceFile:35)
	at net.minecraft.world.biome.BiomeColorHelper.func_180286_a(SourceFile:47)
	at squeek.veganoption.blocks.BlockJutePlant$ColorHandler.func_186720_a(BlockJutePlant.java:240)
	at net.minecraft.client.renderer.color.BlockColors.func_186724_a(BlockColors.java:167)
	at net.minecraft.client.particle.ParticleDigging.func_187154_b(SourceFile:52)
	at net.minecraft.client.particle.ParticleDigging.func_174845_l(SourceFile:47)
	at net.minecraft.client.particle.ParticleDigging$Factory.func_178902_a(SourceFile:105)
	at net.minecraft.client.particle.ParticleManager.func_178927_a(ParticleManager.java:135)
	at net.minecraft.client.renderer.RenderGlobal.func_174974_b(RenderGlobal.java:2132)

-- Particle being added --
Details:
	ID: 37
	Parameters: [I@7a28eb14
	Position: -85.19,63.52,208.69 - World: (-86,63,208), Chunk: (at 10,3,0 in -6,13; contains blocks -96,0,208 to -81,255,223), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
Stacktrace:
	at net.minecraft.client.renderer.RenderGlobal.func_180442_a(RenderGlobal.java:2086)
	at net.minecraft.world.World.func_175720_a(World.java:1092)
	at net.minecraft.world.World.func_175688_a(World.java:1079)
	at net.minecraft.entity.Entity.func_174808_Z(Entity.java:1146)
	at net.minecraft.entity.Entity.func_174830_Y(Entity.java:1132)
	at net.minecraft.entity.Entity.func_70030_z(Entity.java:436)
	at net.minecraft.entity.EntityLivingBase.func_70030_z(EntityLivingBase.java:216)
	at net.minecraft.entity.Entity.func_70071_h_(Entity.java:359)
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:1932)
	at net.minecraft.entity.player.EntityPlayer.func_70071_h_(EntityPlayer.java:233)
	at net.minecraft.client.entity.EntityPlayerSP.func_70071_h_(EntityPlayerSP.java:167)
	at net.minecraft.world.World.func_72866_a(World.java:1964)
	at net.minecraft.world.World.func_72870_g(World.java:1934)

-- Entity being ticked --
Details:
	Entity Type: null (net.minecraft.client.entity.EntityPlayerSP)
	Entity ID: 935
	Entity Name: milchLoser
	Entity's Exact location: -85.16, 63.42, 208.58
	Entity's Block location: World: (-86,63,208), Chunk: (at 10,3,0 in -6,13; contains blocks -96,0,208 to -81,255,223), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Entity's Momentum: 0.26, 0.33, -0.04
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at net.minecraft.world.World.func_72939_s(World.java:1750)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityPlayerSP['milchLoser'/935, l='MpServer', x=-85.16, y=63.42, z=208.58]]
	Chunk stats: MultiplayerChunkCache: 392, 392
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (-20,64,248), Chunk: (at 12,4,8 in -2,15; contains blocks -32,0,240 to -17,255,255), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Level time: 1278 game time, 1278 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 106 total; [EntityMinecartChest['Minecart with Chest'/27, l='MpServer', x=-158.50, y=27.06, z=279.50], EntityMinecartChest['Minecart with Chest'/28, l='MpServer', x=-167.50, y=37.06, z=287.50], EntityBat['Bat'/4127, l='MpServer', x=-24.92, y=28.22, z=179.28], EntityChicken['Chicken'/31, l='MpServer', x=-148.50, y=72.00, z=268.50], EntityChicken['Chicken'/32, l='MpServer', x=-145.50, y=73.00, z=271.50], EntityChicken['Chicken'/33, l='MpServer', x=-142.50, y=73.00, z=272.50], EntityChicken['Chicken'/34, l='MpServer', x=-128.45, y=71.00, z=160.49], EntitySheep['Sheep'/35, l='MpServer', x=-128.50, y=66.00, z=171.50], EntitySheep['Sheep'/36, l='MpServer', x=-127.50, y=65.00, z=172.50], EntityPig['Pig'/37, l='MpServer', x=-130.50, y=64.00, z=184.50], EntityPig['Pig'/38, l='MpServer', x=-130.50, y=63.00, z=192.50], EntityPig['Pig'/39, l='MpServer', x=-120.55, y=64.00, z=191.45], EntityCreeper['Creeper'/4181, l='MpServer', x=-118.50, y=35.00, z=272.50], EntityZombie['Zombie'/4182, l='MpServer', x=-117.50, y=35.00, z=271.50], EntitySheep['Sheep'/92, l='MpServer', x=-57.50, y=97.00, z=262.50], EntityZombie['Zombie'/4188, l='MpServer', x=-62.65, y=13.00, z=247.35], EntitySheep['Sheep'/93, l='MpServer', x=-56.50, y=75.00, z=263.50], EntityZombie['Zombie'/4189, l='MpServer', x=-61.50, y=13.00, z=246.50], EntitySheep['Sheep'/94, l='MpServer', x=-56.50, y=75.00, z=263.50], EntityZombie['Zombie'/4190, l='MpServer', x=-61.50, y=13.00, z=250.50], EntitySheep['Sheep'/95, l='MpServer', x=-62.27, y=76.00, z=265.53], EntitySheep['Sheep'/100, l='MpServer', x=-43.50, y=69.00, z=170.50], EntitySheep['Sheep'/101, l='MpServer', x=-40.50, y=69.00, z=174.50], EntitySheep['Sheep'/102, l='MpServer', x=-34.72, y=71.00, z=178.53], EntitySheep['Sheep'/103, l='MpServer', x=-43.50, y=69.00, z=170.50], EntitySheep['Sheep'/106, l='MpServer', x=-40.50, y=72.00, z=288.50], EntitySheep['Sheep'/107, l='MpServer', x=-31.50, y=72.00, z=159.50], EntitySheep['Sheep'/108, l='MpServer', x=-28.50, y=71.00, z=157.50], EntitySheep['Sheep'/109, l='MpServer', x=-29.50, y=72.00, z=159.50], EntitySheep['Sheep'/110, l='MpServer', x=-29.50, y=72.00, z=160.50], EntitySheep['Sheep'/111, l='MpServer', x=-36.53, y=65.00, z=207.21], EntitySheep['Sheep'/112, l='MpServer', x=-40.40, y=62.22, z=202.89], EntitySheep['Sheep'/113, l='MpServer', x=-35.48, y=62.26, z=199.87], EntitySheep['Sheep'/114, l='MpServer', x=-34.45, y=63.00, z=203.62], EntitySheep['Sheep'/124, l='MpServer', x=-10.50, y=70.00, z=150.50], EntitySheep['Sheep'/125, l='MpServer', x=-8.50, y=70.00, z=148.50], EntitySheep['Sheep'/126, l='MpServer', x=-12.77, y=70.00, z=144.52], EntitySheep['Sheep'/129, l='MpServer', x=-14.27, y=66.00, z=214.51], EntitySheep['Sheep'/130, l='MpServer', x=-22.46, y=65.00, z=207.73], EntitySquid['Squid'/2193, l='MpServer', x=-94.60, y=61.24, z=195.48], EntityCreeper['Creeper'/4265, l='MpServer', x=-133.50, y=19.00, z=243.50], EntitySkeleton['Skeleton'/4308, l='MpServer', x=-72.50, y=26.00, z=263.50], EntitySkeleton['Skeleton'/4309, l='MpServer', x=-75.50, y=26.00, z=265.50], EntitySkeleton['Skeleton'/4314, l='MpServer', x=-44.50, y=13.00, z=132.50], EntitySkeleton['Skeleton'/4315, l='MpServer', x=-46.50, y=13.00, z=130.50], EntitySkeleton['Skeleton'/4316, l='MpServer', x=-44.50, y=13.00, z=131.50], EntityCreeper['Creeper'/4320, l='MpServer', x=-50.50, y=25.00, z=225.50], EntityCreeper['Creeper'/4357, l='MpServer', x=-90.50, y=12.00, z=141.50], EntityBat['Bat'/4412, l='MpServer', x=-36.11, y=46.31, z=237.58], EntityCreeper['Creeper'/4437, l='MpServer', x=-157.50, y=44.00, z=165.50], EntityCreeper['Creeper'/4438, l='MpServer', x=-159.50, y=44.00, z=166.50], EntityCreeper['Creeper'/4439, l='MpServer', x=-160.50, y=44.00, z=166.50], EntityCreeper['Creeper'/4467, l='MpServer', x=-119.50, y=13.00, z=184.50], EntityZombie['Zombie'/4508, l='MpServer', x=-39.50, y=32.00, z=267.50], EntitySkeleton['Skeleton'/4520, l='MpServer', x=-48.50, y=27.00, z=288.50], EntityZombie['Zombie'/4548, l='MpServer', x=-136.50, y=16.00, z=167.50], EntitySkeleton['Skeleton'/4668, l='MpServer', x=-165.50, y=19.00, z=271.50], EntityCreeper['Creeper'/4669, l='MpServer', x=-48.15, y=49.00, z=241.47], EntityBat['Bat'/4749, l='MpServer', x=-149.48, y=49.78, z=162.73], EntityBat['Bat'/4757, l='MpServer', x=-159.05, y=38.08, z=285.42], EntityBat['Bat'/4762, l='MpServer', x=-12.44, y=36.13, z=272.00], EntityZombie['Zombie'/4799, l='MpServer', x=-62.50, y=27.00, z=270.50], EntityBat['Bat'/4829, l='MpServer', x=-69.25, y=14.40, z=244.25], EntityItem['item.item.seeds'/4907, l='MpServer', x=-36.85, y=62.00, z=202.96], EntitySkeleton['Skeleton'/4914, l='MpServer', x=-98.50, y=25.00, z=264.50], EntitySpider['Spider'/4915, l='MpServer', x=-100.50, y=25.00, z=264.50], EntityOcelot['Ocelot'/4918, l='MpServer', x=-164.47, y=70.00, z=266.92], EntityOcelot['Ocelot'/4919, l='MpServer', x=-163.74, y=70.00, z=263.65], EntityOcelot['Ocelot'/4920, l='MpServer', x=-164.85, y=70.00, z=262.15], EntityOcelot['Ocelot'/4921, l='MpServer', x=-164.40, y=70.00, z=262.60], EntitySkeleton['Skeleton'/4952, l='MpServer', x=-56.50, y=44.00, z=249.50], EntityCreeper['Creeper'/4954, l='MpServer', x=-77.50, y=30.00, z=277.50], EntityCreeper['Creeper'/4955, l='MpServer', x=-41.50, y=49.00, z=244.50], EntityZombie['Zombie'/4956, l='MpServer', x=-45.50, y=49.00, z=245.50], EntitySquid['Squid'/3192, l='MpServer', x=-69.62, y=62.01, z=200.32], EntitySquid['Squid'/3206, l='MpServer', x=-102.19, y=59.00, z=206.12], EntitySquid['Squid'/3207, l='MpServer', x=-96.44, y=62.55, z=206.32], EntitySquid['Squid'/3208, l='MpServer', x=-97.57, y=60.95, z=200.82], EntitySquid['Squid'/3209, l='MpServer', x=-90.42, y=62.54, z=197.77], EntityBat['Bat'/3242, l='MpServer', x=-51.25, y=47.10, z=248.21], EntityZombie['Zombie'/3247, l='MpServer', x=-73.50, y=47.00, z=279.50], EntityBat['Bat'/3261, l='MpServer', x=-71.46, y=15.10, z=239.62], EntityBat['Bat'/3302, l='MpServer', x=-36.17, y=43.10, z=287.47], EntityZombie['Zombie'/3310, l='MpServer', x=-66.50, y=20.00, z=131.50], EntityBat['Bat'/3347, l='MpServer', x=-28.06, y=22.07, z=187.03], EntityZombie['Zombie'/3400, l='MpServer', x=-65.50, y=13.00, z=249.50], EntitySkeleton['Skeleton'/3487, l='MpServer', x=-56.50, y=57.00, z=250.50], EntityPlayerSP['milchLoser'/935, l='MpServer', x=-85.16, y=63.42, z=208.58], EntityCreeper['Creeper'/3565, l='MpServer', x=-38.50, y=35.00, z=189.50], EntityZombie['Zombie'/3599, l='MpServer', x=-68.30, y=28.00, z=207.56], EntityZombie['Zombie'/3600, l='MpServer', x=-67.51, y=30.00, z=205.80], EntityCreeper['Creeper'/3603, l='MpServer', x=-47.50, y=29.00, z=216.50], EntityCreeper['Creeper'/3642, l='MpServer', x=-74.83, y=27.00, z=265.48], EntityCreeper['Creeper'/3697, l='MpServer', x=-29.50, y=45.00, z=282.50], EntitySkeleton['Skeleton'/3724, l='MpServer', x=-37.50, y=41.00, z=240.50], EntitySkeleton['Skeleton'/3725, l='MpServer', x=-34.50, y=41.00, z=238.50], EntitySkeleton['Skeleton'/3751, l='MpServer', x=-12.50, y=37.00, z=269.50], EntityWitch['Witch'/3826, l='MpServer', x=-80.50, y=16.00, z=256.71], EntityItem['item.item.seeds'/3835, l='MpServer', x=-40.71, y=63.00, z=208.57], EntityCreeper['Creeper'/3892, l='MpServer', x=-157.50, y=66.00, z=238.50], EntityBat['Bat'/3934, l='MpServer', x=-37.25, y=34.07, z=271.75], EntityBat['Bat'/3935, l='MpServer', x=-34.05, y=33.53, z=269.91], EntitySkeleton['Skeleton'/3973, l='MpServer', x=-21.50, y=26.00, z=182.50], EntityZombie['Zombie'/3978, l='MpServer', x=-52.50, y=18.00, z=132.50], EntitySkeleton['Skeleton'/4035, l='MpServer', x=-48.50, y=45.00, z=254.50], EntityCreeper['Creeper'/4083, l='MpServer', x=-127.50, y=25.00, z=181.50]]
	Retry entities: 0 total; []
	Server brand: fml,forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:415)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2660)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:392)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.10.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_25, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 692103016 bytes (660 MB) / 2691170304 bytes (2566 MB) up to 6049234944 bytes (5769 MB)
	JVM Flags: 4 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx6489m -Xms256m -XX:PermSize=256m
	IntCache: cache: 0, tcache: 0, allocated: 12, tallocated: 94
	FML: MCP 9.32 Powered by Forge 12.18.3.2185 4 mods loaded, 4 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
	UCHIJAAAA	mcp{9.19} [Minecraft Coder Pack] (minecraft.jar) 
	UCHIJAAAA	FML{8.0.99.99} [Forge Mod Loader] (forge-1.10.2-12.18.3.2185.jar) 
	UCHIJAAAA	Forge{12.18.3.2185} [Minecraft Forge] (forge-1.10.2-12.18.3.2185.jar) 
	UCHIJAAAA	VeganOption{0.2.0} [The Vegan Option] (VeganOption-mc1.10.2-0.2.0.jar) 
	Loaded coremods (and transformers): 
ASMPlugin (VeganOption-mc1.10.2-0.2.0.jar)
  squeek.veganoption.asm.ClassTransformer
	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.5.0 NVIDIA 382.05' Renderer: 'GeForce GTX 960/PCIe/SSE2'
	Launched Version: forge-12.18.3.2185
	LWJGL: 2.9.4
	OpenGL: GeForce GTX 960/PCIe/SSE2 GL version 4.5.0 NVIDIA 382.05, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 4x Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz
```

**Comment by elifoster** [comment]
Running the latest Forge for 1.10.2 (12.18.3.2316) I am not able to reproduce this.

**Comment by squeek502** [comment]
Looks like the vanilla IBlockColor implementations do null checks on the `@Nullable` parameters. Might want to just put that in as a precaution? Seems like the BlockPos is what's null in the crash log.

**Comment by squeek502** [comment]
Should be fixed in v0.2.1: https://github.com/squeek502/VeganOption/releases/tag/v0.2.1
